### PR TITLE
Bloom filter merge

### PR DESCRIPTION
**Pull request by faradaym**
**Describe at a high level the solution you're providing**
Based off of #252 , adding in refactored bloomfilter:

- Fixed bloomfilter math
- Refactored bloomfilter
- added acceptance test

**Is this a patch, a minor version change, or a major version change**
major

**Additional context**
Currently:
total number of bits = 10,000 
maximum messages = 100
chance of collision is 1%

### CODE REVIEW

File: src/bloomfilter.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    delete this->filter1;
+}
+
+void BloomFilter::set_hash_results(unsigned char* msg, int msgSize,
+    std::unique_ptr<unsigned int[]> & hashResults
+) {
+    for (int i = 0; i < this->numHashes; i++){
+        unsigned int hashResult = djb2Hash(msg, this->Seeds[i], msgSize);
+        unsigned int totalBitSize = this->numSectors * this->bitsPerSector;
+        hashResult = hashResult % totalBitSize;
```
Comment (rolsen) [inline src/bloomfilter.cpp:89]: This is the place we fixed the bloom filter math. The previous version was something like:

```
hashResult = hashResult % this->numSectors;
```

Terminology:

We're using "sector" to mean one `unsigned int`.

The previous code for this part was basically treating each sector (unsigned int) as a bit map of size one. Other parts of the code treat each sector as a bit map of multiple bits. (The number of bits per sector is the third parameter of the BloomFilter constructor.) Using all the bits in a sector means basically doing `sizeof(unsigned int)` to see how many bytes are in an unsigned int. Then multiply by the number of bits per byte, and you've got the maximum number of bits in a sector. That's 32-bits on my Heltec V2.

Comment (TimoWielink) [reply src/bloomfilter.cpp:89]: Testing this branch now

**Comment by TimoWielink** [comment]
Getting great test results, no collision, and all data seems to be arriving. 

I ran 6 ducks for over an hour with a 10-second interval and runs smooth.

Did not get any hopping going, the duck seems to have good range.

**Comment by nfeuer** [comment]
So you wont see hopping like you did in the past since there is no more path. Thought the hop count should update. Not sure if we push that data to the cloud

File: src/bloomfilter.cpp (hunk 1 above)
Comment (nfeuer) [reply src/bloomfilter.cpp:89]: Would a hardware running on an 8-bit system work differently or will it adust?

**Comment by nfeuer** [comment]
@faradaym can you update the version to 3.0.1?

File: src/bloomfilter.cpp (hunk 1 above)
Comment (faradaym) [reply src/bloomfilter.cpp:89]: Currently, it is hard-coded. We could use `sizeof()` to get the size based on the device's architecture.

**Pull request merged**